libsolidity/analysis/TypeChecker.cpp

```diff
@@ -592,7 +592,12 @@ bool TypeChecker::visit(VariableDeclaration const& _variable)
 				);
 		}
 		if (!getter.interfaceFunctionType())
-			m_errorReporter.typeError(6744_error, _variable.location(), "Internal or recursive type is not allowed for public state variables.");
+		{
+			if (getter.returnParameterNames().empty() && getter.returnParameterTypes().empty() && getter.parameterNames().empty() && getter.parameterTypes().empty())
+				m_errorReporter.typeError(5359_error, _variable.location(), "The structure has all it's members omitted, therefore the getter cannot return any values.");
```

Comment:
I think you can assume that there are as many names as there are types:
```suggestion
			solAssert(getter.returnParameterNames().size() == getter.returnParameterTypes().size());
			solAssert(getter.parameterNames().size() == getter.parameterTypes().size());
			if (getter.returnParameterTypes().empty() && getter.parameterTypes().empty())
				m_errorReporter.typeError(5359_error, _variable.location(), "The structure has all it's members omitted, therefore the getter cannot return any values.");
```
Or might be `<=` - not sure if in case like `function f(uint x, uint)` you'll get less names or just `""` as names. Please see if tests will pass with one of these.